packages/calimero-sdk/src/subscriptions/subscriptions.ts

```diff
@@ -1,10 +1,10 @@
 import { ContextId } from '../types/context';
 
 export interface SubscriptionsClient {
-  connect(connectionId: string): void;
-  disconnect(connectionId: string): void;
-  subscribe(applicationIds: string[], connectionId?: string): void;
-  unsubscribe(applicationIds: string[], connectionId?: string): void;
+  connect(connectionId?: string): Promise<void>;
```

Comment:
Major issue was that subscription didn't return promise so app didn't wait for connection